pkg/mdl/factory.go

```diff
@@ -85,3 +88,29 @@ func EmptyUintIfNil(i *uint) uint {
 func Time(t time.Time) *time.Time {
 	return &t
 }
+
+func IsNil(m interface{}) bool {
+	if m == nil {
+		return true
+	}
+
+	v := reflect.ValueOf(m)
+
+	switch v.Kind() {
+	case reflect.Chan,
+		reflect.Func,
+		reflect.Map,
+		reflect.Ptr,
+		reflect.UnsafePointer,
+		reflect.Interface,
+		reflect.Slice:
+	default:
+		return true
```

Comment:
Shouldn't this be `return false`? If you add some tests for this, I would expect that an empty struct is not nil, but I would also expect the current implementation would say it is nil.

Reply:
i would rather want to check it properly, but what kind of types should i add checks for? don't want to make this a 100 liner